### ISSUE TITLE
Request: Ability to turn the fingerprint LED off completely

### ISSUE THREAD

**Issue by tangowithfoxtrot** [FrameworkComputer/framework-system#37]
The fingerprint/power LED can be controlled thanks to the work done in #9. Being able to set this to "low" is very much appreciated, but it would be nice to have to option to turn it off completely.

Is disabling the LED completely possible?

**Comment by JohnAZoidberg** [comment]
That's an interesting idea.
But one of the purposes of the power button LED is to let you know whether the system is awake or asleep.

When the system is asleep/suspended the power button LED shows a breathing animation. If it's always off, you can't distinguish between the system being asleep or just screen off.

Please help me understand the reason and situation in which you'd like to have the LED turned off completely. Maybe we can figure out a good solution.

**Comment by tangowithfoxtrot** [comment]
The reason I'd like to have the LED completely off is that in the dark, even low brightness is pretty distracting and un-easy on my eyes, particularly when using dark themes or watching movies, where you generally don't want other sources of light interfering with your viewing experience. I usually disable the keyboard backlight as well in those situations, so having a light that can't be disabled there is a bit of a bummer.

The suspend situation you described is definitely a trade-off though. For me, it's one I'd be willing to make, as it makes my general usage much more pleasant, and my laptop is set to wake from suspend when I open the lid, so I don't have to worry about the indicator being a source of truth for that. I figure that mostly anyone using the input module CLIs would be pretty aware of any configurations that they've done in that regard and adapt their expectations for how the light behaves accordingly.

Either way, thanks for your response and consideration!

**Comment by tangowithfoxtrot** [comment]
Alternatively, if an "off" mode is considered to be too niche of a request, maybe offer an "ultra-low" mode that illuminates the LED at the lowest possible value it can be while still being on. This would be a compromise for me, as I'd still like to remove any extraneous distractions from the screen when using it in low-light environments, but it's an improvement.

Another alternative might be to allow adjusting by percentage values and letting people set it to 0%, but I'm not sure if the underlying firmware supports that or how much additional work would be involved in making that work.

**Comment by JohnAZoidberg** [comment]
We will introduce a new ultra-low option in the BIOS settings and also allow you to set it to a custom percentage down to 1%.

**Comment by JohnAZoidberg** [comment]
> We will introduce a new ultra-low option in the BIOS settings and also allow you to set it to a custom percentage down to 1%.

Ultra low will be 7% PWM, it's the lowest that still has a nice breathing animation in suspend.

**Comment by tangowithfoxtrot** [comment]
Thanks for the updates and considering my request!

Is the code that controls what the minimum-allowed brightness something that I can change myself, or is that part of the UEFI firmware itself? I assume if it's the latter, it would be controlled by code that I do not have access to?

1% seems like it'd be minimally distracting in dark viewing scenarios, but I'm less sure about 7%. It would be nice if I could set that to a value lower than 7% (or maybe even off), even if I have to fork and build myself.

**Comment by JohnAZoidberg** [comment]
See #85, as I mentioned, you'll be able to manually anything inbetween 1%-100% brightness.
As previously it's overriden by the BIOS during boot, so you an create a startup script in linux.

Or you could use the UEFI version of this tool and run it during boot by your bootloader.

Or you could (already) modify and build your own EC firmware: https://github.com/frameworkComputer/embeddedcontroller
But that's obviously the riskiest choice, flashing your EC firmware.
If that's broken, you need special hardware to recover your firmware and unbrick your system.

**Comment by tangowithfoxtrot** [comment]
Ah okay. Thanks for clarifying. I didn't know there was a distinction between the "ultra-low" option in the UEFI and the actual minimum-allowed brightness you can set manually. As long as the `framework_tool` CLI can set it to anything between 1 and 100%, that's perfect!

Thanks again!

**Comment by tripplehelix** [comment]
Not able to set the 7040 Series Framework13 BIOS 03.07 fingerprint light to ultra-low.

`[ERROR] EC Response Code: InvalidParameter`

Am I too quick, and it's not available yet? 

Also, `--fp-brightness` does nothing, just get `InvalidVersion`.

**Comment by JohnAZoidberg** [comment]
It needs updated bios that's not available yet

**Comment by kariudo** [comment]
I would still like to see the option to set it off (0%) when desired. Is there really a good reason to not allow it? I understand there is valid reason to have it on to show status; however, I would argue that pressing any key will quickly tell you if the system was suspended since it will wake up if it was. I think being able to use the side LED would be a reasonable option for showing a "breathing" when sleeping.

It seems unfortunate to restrict it completely when there is a desire to be able to have it completely dark (personally I like my keyboard unilluminated and monitor at minimum brightness when I'm using my laptop in the dark, as I'm doing right now. I'd rather in that scenario that the light not be there to be distracting.

Even if most people might want the default of leaving it on auto, is there harm in allowing manual selection of off?

**Comment by t-8ch** [comment]
> [kiram9](https://github.com/kiram9)
added 3 commits that reference this issue [last week](https://github.com/FrameworkComputer/framework-system/issues/37#event-21629529751)

> [fwk/main: EC_CMD_FP_LED_LEVEL_CONTROL v1 set percentage](https://github.com/FrameworkComputer/EmbeddedController/commit/72ace9e0aeb4f8810dfe224dfd14a74b081001ea)

The standard command `EC_CMD_LED_CONTROL` already supports fractional brightness levels.
Why not make that work instead?

**Comment by kirelagin** [comment]
Yes, I don’t understand this too: it should be pretty straightforward to just make `ectool led power white=0x01` work, right?

**Comment by Qukell** [comment]
So is there a way to turn off the power button / fingerprint LED on Windows by now? It is extremely annoying, distracting, and unnecessary when it is lit up in the dark - after all, the keyboard backlighting can be turned off as well.
